lib/executor/ramping_arrival_rate.go

```diff
@@ -479,10 +480,7 @@ func (varr RampingArrivalRate) Run(parentCtx context.Context, out chan<- stats.S
 		// Since there aren't any free VUs available, consider this iteration
 		// dropped - we aren't going to try to recover it, but
 
-		stats.PushIfNotDone(parentCtx, out, stats.Sample{
-			Value: 1, Metric: metrics.DroppedIterations,
-			Tags: metricTags, Time: time.Now(),
-		})
+		droppedIterationMetric.Emit(parentCtx, time.Now(), metricTags, 1)
```

Comment:
This seems a lot better to me than the above or the Sample.Push I show in the `constant-arrival-rate`
https://github.com/k6io/k6/pull/2071/files#diff-2b3a85b9dccbd5050f5755479947c5237340bf7335dba7f83d654c55548df710R349-R352

Reply:
Yeah, I saw this, and it's slightly better than `Sample.Push()`, but I still think we're not separating concerns properly